velox/exec/fuzzer/FuzzerUtil.cpp

```diff
@@ -36,6 +52,37 @@ void writeToFile(
   writer.close();
 }
 
+// Recursive function to create splits with their corresponding schemas and
+// store in splits.
+// In a table directory, each partition would be stored as a
+// sub-directory, multiple partition columns would make up nested directory
+// structure.
+//
+// For example for a file path such as /p0=0/p1=0/0000_file1, creates
+// split with partition keys (p0, 0), (p1 0)
+void makeSplitsWithSchema(
+    std::vector<Split>& splits,
+    const std::string& directory,
+    std::unordered_map<std::string, std::optional<std::string>> partitionKeys) {
+  for (auto const& entry : std::filesystem::directory_iterator{directory}) {
+    if (entry.is_directory()) {
+      auto directoryName = entry.path().string();
+      auto partition =
+          extractPartition(directoryName.substr(directory.size() + 1));
+      partitionKeys.insert(
+          {partition.first,
+           partition.second == DEFAULT_PARTITION_VALUE
+               ? std::nullopt
+               : std::optional(partition.second)});
+      makeSplitsWithSchema(splits, directoryName, partitionKeys);
+      partitionKeys.erase(partition.first);
```

Comment:
Why erase?

Reply:
This is in a for-loop, for example under /p0=0 folder, current partitionKeys would be (p0=0) we have two folders: 
/p0=0/p1=0/file1     insert (p1=0), partitionKeys become (p0=0, p1=0) and create split, then we have to remove p1=0 
/p0=0/p1=1/file2     so that here we can insert (p1=1) partitionKeys become (p0=0, p1=1)

Reply:
I see. Thanks!